flash/image/classification/integrations/baal/loop.py

```diff
@@ -0,0 +1,162 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+from functools import partial
+from typing import Any, Callable, Dict, Optional
+
+import torch
+from pytorch_lightning.loops import Loop
+from pytorch_lightning.loops.fit_loop import FitLoop
+from pytorch_lightning.trainer.connectors.data_connector import _PatchDataLoader
+from pytorch_lightning.trainer.progress import Progress
+from pytorch_lightning.trainer.states import RunningStage, TrainerFn, TrainerStatus
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from flash.core.utilities.imports import requires
+from flash.image.classification.integrations.baal.data import ActiveLearningDataModule
+
+
+class ActiveLearningLoop(Loop):
+    @requires("baal")
+    def __init__(self, label_epoch_frequency: int, inference_iteration: int = 2):
+        """The `ActiveLearning Loop` describes the following training procedure. This loop is connected with the
+        `ActiveLearningTrainer`
+
+        Example::
+
+            while unlabelled data or budget critera not reached:
+
+                if labelled data
+                    trainer.fit(model, labelled data)
+
+                if unlabelled data:
+                    predictions = trainer.predict(model, unlabelled data)
+                    uncertainties = heuristic(predictions)
+                    request labellelisation for the sample with highest uncertainties under a given budget
+
+        Args:
+            label_epoch_frequency: Number of epoch to train on before requesting labellisation.
+            inference_iteration: Number of inference to perform to compute uncertainty.
+        """
+        super().__init__()
+        self.label_epoch_frequency = label_epoch_frequency
+        self.inference_iteration = inference_iteration
+        self.fit_loop: Optional[FitLoop] = None
+        self.progress = Progress()
+        self._should_continue: bool = False
+        self._model_state_dict: Optional[Dict[str, torch.Tensor]] = None
+
+    @property
+    def done(self) -> bool:
+        return self.progress.current.completed > self.max_epochs or self._should_continue
```

Comment:
```suggestion
        return self.progress.current.completed > self.max_epochs or not self._should_continue
```

Reply:
Done ! Good catch !

Reply:
I removed it as it is not used right now.